### PR TITLE
Add rollback function and migrate only selected tables feature

### PR DESCRIPTION
**Pull request by NeerajGartia21**
This PR adds rollback feature if any error occurs and also user can select the tables to migrate

### CODE REVIEW

**Comment by NeerajGartia21** [comment]
@chivalryq Can you please review this PR.

File: README.md [hunk 1]
```diff
@@ -1 +1,50 @@
...
+  URL: "<user>:<password>d@<host>/<database>"
+  Type: "mysql"
+  Database: "kubevela"
+
+target:
+  URL: "mongodb+srv://<username>:<password>@<host>"
+  Type: "mongodb"
+  Database: "kubevela"
+
+ErrorOnDup: "update"
```
Comment (chivalryq) [inline README.md]: ErrorOnDup is not snake case. It's wierd in this config.

Also, add possible options in yaml comments.

File: pkg/cmd/command.go [hunk 2]
```diff
@@ -18,13 +21,26 @@ func NewMigratorCommand() *cobra.Command {
...
-			return nil
-		},
+	}
+	cmd.RunE = func(cmd *cobra.Command, args []string) error {
+		if len(args) > 1 {
+			return fmt.Errorf("too many arguments")
+		}
+		if len(args) == 0 {
+			return fmt.Errorf("provide the path of the config file")
+		}
```
Comment (chivalryq) [inline pkg/cmd/command.go]: It would be better to use `-c config.yaml` style instead of position argument.

File: pkg/cmd/command.go [hunk 3]
```diff
@@ -18,13 +21,26 @@ func NewMigratorCommand() *cobra.Command {
...
+		config, err := utils.LoadConfig(args[0])
+		if err != nil {
+			return err
+		}
+		err = migrateCmd(cmd.Context(), config)
+		if err != nil {
+			return err
+		}
+		return nil
```
Comment (chivalryq) [inline pkg/cmd/command.go]: just `return migrateCmd(cmd.Context(), config)`

File: pkg/cmd/helper.go [hunk 4]
```diff
@@ -0,0 +1,155 @@
...
+		updateOnDup := strings.ToLower(actionOnDup) == types.UpdateOnDup
+		var tableMigrated = false
+		rows, err := source.List(ctx, table, nil)
+		if err != nil {
+			fmt.Printf("Error migrating %s table\n", table.TableName())
+			errOnRollback := Rollback(ctx, target, tables, deleteTables, restoreTables)
+			if errOnRollback != nil {
+				return errOnRollback
+			}
+			return err
```
Comment (chivalryq) [inline pkg/cmd/helper.go]:
```suggestion
rollback:=func()error{
			errOnRollback := Rollback(ctx, target, tables, deleteTables, restoreTables)
			if errOnRollback != nil {
				return errOnRollback
			}
			return err
}

// when an error occurs in migragating, just use `rollback()`
```

File: pkg/cmd/helper.go [hunk 5]
```diff
@@ -0,0 +1,155 @@
...
+					if errOnRollback != nil {
+						return errOnRollback
+					}
+					return err
+				}
+			}
+			deleteTables[saveEntity.TableName()] = append(deleteTables[saveEntity.TableName()], initialEntity)
+		}
+		if tableMigrated {
+			fmt.Printf("Migrated table %s\n", table.TableName())
```
Comment (chivalryq) [inline pkg/cmd/helper.go:142]: How about print a statistic info like how many entry is migrated. So user can get better control of this process.

How many entry:
- is traversed (number of entries in source)
- is copied
- is update or ignore on duplication

File: pkg/cmd/helper.go [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+							err := Rollback(ctx, target, tables, deleteTables, restoreTables)
+							if err != nil {
+								return err
+							}
+							return err
+						}
+						restoreTables[saveEntity.TableName()] = append(restoreTables[saveEntity.TableName()], initialEntity)
+						continue
+					}
+					fmt.Printf("Error migrating %s table\n", table.TableName())
```
Comment (chivalryq) [inline pkg/cmd/helper.go:132]: Please don't swallow the error message. If errOnRollback, then err is forever lost.

File: README.md (hunk 1 above)
Comment (NeerajGartia21) [reply README.md]: Is camel case a good choice?

Comment (NeerajGartia21) [reply README.md]: Also I have added the possible options in a note, but I will add them in the yaml also.

File: pkg/cmd/command.go (hunk 2 above)
Comment (NeerajGartia21) [reply pkg/cmd/command.go]: Yeah. It's better to use the -c flag.

File: pkg/cmd/helper.go (hunk 5 above)
Comment (NeerajGartia21) [reply pkg/cmd/helper.go:142]: Yeah. I was also thinking to implement something like that in the next PR.

File: README.md (hunk 1 above)
Comment (chivalryq) [reply README.md]: Sorry I mean it's not camel case. So it is inconsistent with other fields.

File: pkg/cmd/helper.go (hunk 6 above)
Comment (NeerajGartia21) [reply pkg/cmd/helper.go:132]: Oh sorry. I missed that.

File: pkg/cmd/command.go (hunk 2 above)
Comment (NeerajGartia21) [reply pkg/cmd/command.go]: Done!!

File: pkg/cmd/command.go (hunk 3 above)
Comment (NeerajGartia21) [reply pkg/cmd/command.go]: Done !!

File: pkg/cmd/helper.go (hunk 4 above)
Comment (NeerajGartia21) [reply pkg/cmd/helper.go]: Done !!

File: README.md (hunk 1 above)
Comment (NeerajGartia21) [reply README.md]: Changed the `ErrorOnDup` to `actionOnDup` in camelCase

File: pkg/cmd/helper.go (hunk 6 above)
Comment (NeerajGartia21) [reply pkg/cmd/helper.go:132]: Done!!

File: README.md (hunk 1 above)
Comment (NeerajGartia21) [reply README.md]: Can you pls suggest a good choice?
I thought camelCase is a good choice as camelcase is used in some config files.

**Review by chivalryq — APPROVED** [review]
LGTM

**Pull request merged**